### PR TITLE
Healing nanites do stamina damage

### PR DESCRIPTION
**Pull request by Mqiib**
# General Documentation

### Intent of your Pull Request

Healing nanites are really, really powerful with almost no downside. This adds a non-dangerous but annoying downside to having them run all the time by causing them to deal stamina damage (up to a maximum) while running.

Lore explanation: 
-The accelerated regen put your natural healing process into overdrive causing wounds to burn and itch. Not only does this hurt, but it puts a huge strain on the body to actually heal the damage causing you to tire out quickly.
-Bio reconstruction nanites are actively ripping apart and reconstructing damaged tissue which causes an immense amount of pain. This pain is great enough that normal people have a hard time dealing with it for any significant length of time.

### Why is this change good for the game?

nanites op, this makes them less op

# Wiki Documentation

### Briefly describe your PR and the impacts of it, in layman's terms. 

Healing nanites will also deal stamina damage up to a limit while healing. Bio recon program will also prevent stamina regen so long as the user is actively benefitting from them when over the stamina damage limit.

### What should players be aware of when it comes to the changes your PR is implementing?

Healing nanites now deal stamina damage while running, up to a lower limit for accelerated regen nanites and a higher limit for bio reconstruction nanites. Bio reconstruction will also prevent stamina from resetting while it is actively healing.

### What general grouping does this PR fall under? 

Science rebalance/tweak

### Are there any aspects of the PR that you would like us not to mention on the Wiki?
No

### If there are any numerical values involved in your PR that will be relevant to a player, please note them here. 
Accelerated regen: 1 stamina damage/process up to 50 (actual 40)
Bio Reconstruction: 5 stamina damage/process up to 100 (actual 80) and will continue to do 0.1/process as long as it's still healing

# Changelog

:cl:  
rscadd: Accelerated regeneration and bio-reconstruction nanites will do stamina damage while active (up to a limit (usually)).

/:cl:

### CODE REVIEW

**Comment by alexkar598** [comment]
Im concerned about the mid combat applications of this ie give someone regen nanites and hurty nanites and watch them stam crit out of nowhere

**Comment by maxion12345** [comment]
what if instead of stamina damage we just like, increased the nanite useage of bioconstruction as to consume massive amounts of nanites, so it'll be good for maybe one "crit to full" heal then you gotta go relax and eat some burgers to regen nanites
 (the only good healing program, accelerated regen while nice isn't going to really keep you alive in a fight and is mostly for post-fight healing or "fuck i had harm intent on" moments)

**Comment by Mqiib** [comment]
> Im concerned about the mid combat applications of this ie give someone regen nanites and hurty nanites and watch them stam crit out of nowhere

They won't stam crit you under most circumstances.

**Comment by Mqiib** [comment]
x

> what if instead of stamina damage we just like, increased the nanite useage of bioconstruction as to consume massive amounts of nanites, so it'll be good for maybe one "crit to full" heal then you gotta go relax and eat some burgers to regen nanites
> (the only good healing program, accelerated regen while nice isn't going to really keep you alive in a fight and is mostly for post-fight healing or "fuck i had harm intent on" moments)

That's boring.

**Comment by SomeguyManperson** [comment]
> Im concerned about the mid combat applications of this ie give someone regen nanites and hurty nanites and watch them stam crit out of nowhere

can't you just kill them with the hurty nanites then

**Comment by Moggram** [comment]
I see this as a good down side. Also alot of you act like there are not chems that recovers stamina.

File: code/modules/research/nanites/nanite_programs/healing.dm [hunk 1]
```diff
@@ -176,6 +180,16 @@
...
+		if(C.getStaminaLoss() < 100) //Stops after hitting the second slowdown level. This is distributed so only ~80% is counted towards crit
+			C.adjustStaminaLoss(5) //Hurts a lot more
+		else if(C.getBruteLoss() || C.getFireLoss()) //Prevents stamina regen if it's actively healing and you're over the limit.
+			C.adjustStaminaLoss(0.1) //Not healing any wounds, so the nanites will reconstruct random healthy tissue. Very painful!
+		if(!C.getBruteLoss() && !C.getFireLoss())
+			if(prob(5))	
+				to_chat(C, "<span class='warning'>You feel a searing pain across your body!")
+		else
+			if(prob(5))
+				to_chat(C, "<span class='warning'>Your wounds burn horribly as they heal!")
```
Comment (alexkar598) [inline code/modules/research/nanites/nanite_programs/healing.dm]:
```suggestion
		if(prob(5))	
			if(!C.getBruteLoss() && !C.getFireLoss())
				to_chat(C, "<span class='warning'>You feel a searing pain across your body!")
			else
				to_chat(C, "<span class='warning'>Your wounds burn horribly as they heal!")
```

File: code/modules/research/nanites/nanite_programs/healing.dm [hunk 2]
```diff
@@ -176,6 +180,16 @@
 				update = TRUE
 		if(update)
 			host_mob.update_damage_overlays()
+		if(C.getStaminaLoss() < 100) //Stops after hitting the second slowdown level. This is distributed so only ~80% is counted towards crit
+			C.adjustStaminaLoss(5) //Hurts a lot more
+		else if(C.getBruteLoss() || C.getFireLoss()) //Prevents stamina regen if it's actively healing and you're over the limit.
+			C.adjustStaminaLoss(0.1) //Not healing any wounds, so the nanites will reconstruct random healthy tissue. Very painful!
```
Comment (alexkar598) [inline code/modules/research/nanites/nanite_programs/healing.dm]: Im not sure what you mean by this comment as this code triggers only when healing wounds

**Review by alexkar598 — DISMISSED** [review]

File: code/modules/research/nanites/nanite_programs/healing.dm (hunk 2 above)
Comment (Mqiib) [reply code/modules/research/nanites/nanite_programs/healing.dm]: whoops, put the comment in the wrong spot

**Comment by boodaliboo** [comment]
you do know accelerated regen automatically activates on its own so people can unknowingly keep getting tons of stamina damage whenever they take any damage

**Comment by Mqiib** [comment]
> you do know accelerated regen automatically activates on its own so people can unknowingly keep getting tons of stamina damage whenever they take any damage

Yes that's how it's supposed to work. If it heals you, you take stamina damage. Since accelerated regen doesn't activate when you are unharmed, it doesnt do stamina damage CONSTANTLY like bio recon

**Comment by boodaliboo** [comment]
intresting then this sounds good, please expand nanites more or port some of tgs prs for it its in desperate need of updating

**Comment by burn1cycle** [comment]
>Healing nanites are really, really powerful with almost no downside. 

Wrong, the downside is that someone can upload malicious programs to the cloud. It would be difficult to convince people to get nanites if the healing ability did this.

File: code/modules/research/nanites/nanite_programs/healing.dm [hunk 3]
```diff
@@ -176,6 +180,15 @@
 				update = TRUE
 		if(update)
 			host_mob.update_damage_overlays()
+		if(C.getStaminaLoss() < 100) //Stops after hitting the second slowdown level. This is distributed so only ~80% is counted towards crit
```
Comment (SomeguyManperson) [inline code/modules/research/nanites/nanite_programs/healing.dm]: damage get procs use % to crit total not actual damage so this will actually stamcrit people

Comment (Mqiib) [reply code/modules/research/nanites/nanite_programs/healing.dm]: 👍

**Review by SomeguyManperson — APPROVED** [review]
